lib/overcommit/hook/pre_commit/line_endings.rb

```diff
@@ -0,0 +1,42 @@
+module Overcommit::Hook::PreCommit
+  # Checks for line endings in files.
+  class LineEndings < Base
+    def run
+      messages = []
+
+      text_files.map do |file_name|
+        file = File.open(file_name)
+        file.each_line do |line|
+          next if unix? && line =~ /\A((?!\r).)*\n\z/
+          next if windows? && line =~ /\A.*\r\n\z/
```

Comment:
I wanted to use positive matching (`line =~ /.*\n/`) instead of negative matching (`!(line =~ /.*\r\n/)`), but for performance it definitely makes more sense to use `end_with?`.